apps/src/code-studio/components/AssetRow.jsx

```diff
@@ -41,6 +45,15 @@ export default class AssetRow extends React.Component {
     attemptedUsedDelete: false
   };
 
+  api = () => {
+    const {api, useFilesApi} = this.props;
+    if (api) {
+      return api;
+    } else {
+      return useFilesApi ? filesApi : assetsApi;
```

Comment:
nit: it's a bit weird to have the api-picking logic split like this. I would recommend moving it entirely outside of (or entirely inside of) The AssetRow. fine to consider doing in a later PR

Reply:
good call! i'll address in a follow-up PR